### PR TITLE
Add AGI investment income exclusion

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
Adds the logic and a test for `invinc_ec_base` from the `taxcalc` [source](https://github1s.com/pslmodels/tax-calculator/blob/HEAD/taxcalc/calcfunctions.py#L401).

### CODE REVIEW

File: openfisca_us/variables/finance/tax/outputs.py [hunk 1]
```diff
@@ -998,9 +999,31 @@ class invinc_ec_base(Variable):
...
     definition_period = YEAR
-    documentation = (
-        """search taxcalc/calcfunctions.py for how calculated and used"""
-    )
+    label = "AGI investment income exclusion"
+    unit = "currency-USD"
+    documentation = """Exclusion of investment income from AGI"""
+
+    def formula(tax_unit, period, parameters):
+        # Limitatition on net short-term and
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:1007]:
```suggestion
        # Limitation on net short-term and
```

File: openfisca_us/variables/finance/tax/outputs.py [hunk 2]
```diff
@@ -998,9 +999,31 @@ class invinc_ec_base(Variable):
...
+        # long-term capital losses
+        limited_capital_gain = max_(
+            -3000.0 / tax_unit("sep", period),
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:1010]: So...I kind of think we should just parameterize/make a new variable for this, given they're doing it in https://github.com/PSLmodels/Tax-Calculator/issues/2624

Could be a separate PR if you prefer though

File: openfisca_us/variables/finance/tax/outputs.py [hunk 3]
```diff
@@ -998,9 +999,31 @@ class invinc_ec_base(Variable):
...
+            add(tax_unit, period, "filer_p22250", "filer_p23250"),
+        )
+        other_inv_income = add(
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:1013]:
```suggestion
        OTHER_INV_INCOME_VARS = ["e00300","e00600","e01100","e01200"]
        other_inv_income = add(
```

File: openfisca_us/variables/finance/tax/outputs.py [hunk 4]
```diff
@@ -998,9 +999,31 @@ class invinc_ec_base(Variable):
...
+            tax_unit,
+            period,
+            *[
+                "filer_" + variable
+                for variable in (
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:1018]:
```suggestion
                for variable in OTHER_INV_INCOME_VARS
```

**Review by MaxGhenis — APPROVED** [review]

**Review by MaxGhenis — COMMENTED** [review]
Could you remind me why we're using `filer_*` variables here?

File: openfisca_us/variables/finance/tax/outputs.py (hunk 1 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/outputs.py:1007]: Fixed

File: openfisca_us/variables/finance/tax/outputs.py (hunk 2 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/outputs.py:1010]: Yep, filed in [#258](https://github.com/PolicyEngine/openfisca-us/issues/258)

**Comment by nikhilwoodruff** [comment]
> Could you remind me why we're using `filer_*` variables here?

Taxdata disregards dependent incomes at the microdata step, and we want to do that in openfisca-us. Effectively all income variables `e123456` will use the filer classes.
